metrics/internal/collectors/cluster-advance-feature-use.go

```diff
@@ -0,0 +1,175 @@
+package collectors
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/red-hat-storage/ocs-operator/metrics/internal/options"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	rookclient "github.com/rook/rook/pkg/client/clientset/versioned"
+	cephv1listers "github.com/rook/rook/pkg/client/listers/ceph.rook.io/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
+
+	storagev1 "k8s.io/api/storage/v1"
+	storagev1listers "k8s.io/client-go/listers/storage/v1"
+)
+
+type ClusterAdvanceFeatureCollector struct {
+	AdvanceFeature      *prometheus.Desc
+	Informer            cache.SharedIndexInformer
+	AllowedNamespaces   []string
+	namespaceClusterMap map[string]string
+	advanceFeatureMap   map[string]int
+}
+
+const (
+	// component within the project/exporter
+	advFeatureSubSystem = "advance_feature"
+)
+
+var _ prometheus.Collector = &ClusterAdvanceFeatureCollector{}
+
+// NewClusterAdvanceFeatureCollector constructs the StorageCluster's advance-feature collector
+func NewClusterAdvanceFeatureCollector(opts *options.Options) *ClusterAdvanceFeatureCollector {
+	client, err := rookclient.NewForConfig(opts.Kubeconfig)
+	if err != nil {
+		klog.Error(err)
+	}
+
+	lw := cache.NewListWatchFromClient(client.CephV1().RESTClient(), "cephclusters", metav1.NamespaceAll, fields.Everything())
+	sharedIndexInformer := cache.NewSharedIndexInformer(lw, &cephv1.CephCluster{}, 0, cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc})
+
+	return &ClusterAdvanceFeatureCollector{
+		AdvanceFeature: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, advFeatureSubSystem, "status"),
+			`Advance features used or not`,
+			[]string{"ceph_cluster", "namespace"},
+			nil,
+		),
+		Informer:            sharedIndexInformer,
+		AllowedNamespaces:   opts.AllowedNamespaces,
+		advanceFeatureMap:   make(map[string]int),
+		namespaceClusterMap: make(map[string]string),
```

Comment:
Can we do this with 1 map? Using "namespace/name" as key and  integer value?
Since we don't care about which features are used, maybe we could save some space?

Reply:
New changes/logic submitted. But overall we use more variables and function calls (more processing), making it more complicated.